platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/project/DeclarativeDSLCustomDependenciesExtensionsSpec.groovy

```diff
@@ -543,8 +543,8 @@ class DeclarativeDSLCustomDependenciesExtensionsSpec extends AbstractIntegration
 
                 """
                 java {
-                    sourceCompatibility = "${lastSupportedVersion}"
-                    targetCompatibility = "${lastSupportedVersion}"
+                    sourceCompatibility = JavaVersion.toVersion("${lastSupportedVersion}")
```

Comment:
This is a sad change, but it requires something like
https://github.com/gradle/gradle/issues/27315
to be implemented. Same problem as with repository.url/buildCache.url